From f0355c52e941586b97d63acff7f706e2bedeba68 Mon Sep 17 00:00:00 2001
From: John Audia <therealgraysky@proton.me>
Date: Thu, 18 Dec 2025 10:10:55 -0500
Subject: [PATCH] kernel: bump 6.12 to 6.12.63

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.63

Manually rebased:
  qualcommax/patches-6.12/0113-remoteproc-qcom-Add-secure-PIL-support.patch

Removed upstreamed:
  airoha/patches-6.12/029-14-spi-airoha-snfi-en7523-workaround-flash-damaging.patch[1]
  qualcommax/patches-6.12/0116-remoteproc-qcom-Update-regmap-offsets-for-halt-regis.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.63&id=7be679104357439ff6eab076975f5f74368acd69
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.63&id=a553e969dc9c41d265c9d46c3e518e424c0b2adb

Build system: x86/64
Build-tested: flogic/glinet_gl-mt6000, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/glinet_gl-mt6000, ramips/tplink_archer-a6-v3, x86/64-glibc
---
 ...oid-reading-flash-page-settings-from.patch |  4 +-
 ...nfi-en7523-workaround-flash-damaging.patch | 97 -------------------
 ...-for-all-the-downstream-rpi-sound-ca.patch | 16 +--
 ...wc3-Set-DMA-and-coherent-masks-early.patch |  4 +-
 ...stb-l2-Add-config-for-2711-controlle.patch |  2 +-
 .../generic/hack-6.12/902-debloat_proc.patch  |  2 +-
 target/linux/generic/kernel-6.12              |  4 +-
 ...ow_offload-handle-netdevice-events-f.patch |  2 +-
 ...moteproc-qcom-Add-secure-PIL-support.patch |  2 +-
 ...Update-regmap-offsets-for-halt-regis.patch | 79 ---------------
 ...c-wcss-disable-auto-boot-for-IPQ8074.patch |  8 +-
 ...wcss-populate-driver-data-for-IPQ601.patch |  8 +-
 ...wcss-change-ssr-name-for-ipq6018-wif.patch |  4 +-
 ...q6v5_wcss-add-optional-qdss_at-clock.patch |  6 +-
 .../850-awake-rt305x-dwc2-controller.patch    |  2 +-
 .../318-add-rtl83xx-clk-support.patch         |  2 +-
 ...chip-Split-up-RK3588-s-PCIe-pinctrls.patch |  4 +-
 ...dts-rockchip-Enable-HDMI0-on-rock-5a.patch |  4 +-
 ...ip-Fix-label-name-of-hdptxphy-for-RK.patch |  2 +-
 ...ip-Add-GPU-power-domain-regulator-de.patch |  2 +-
 ...p-Update-LED-properties-for-Radxa-Ro.patch |  2 +-
 ...05-clk-add-drivers-for-siflower-socs.patch |  2 +-
 22 files changed, 41 insertions(+), 217 deletions(-)
 delete mode 100644 target/linux/airoha/patches-6.12/029-14-spi-airoha-snfi-en7523-workaround-flash-damaging.patch
 delete mode 100644 target/linux/qualcommax/patches-6.12/0116-remoteproc-qcom-Update-regmap-offsets-for-halt-regis.patch

diff --git a/target/linux/airoha/patches-6.12/029-13-spi-airoha-avoid-reading-flash-page-settings-from.patch b/target/linux/airoha/patches-6.12/029-13-spi-airoha-avoid-reading-flash-page-settings-from.patch
index 2efc56f3567704..1822035f27385a 100644
--- a/target/linux/airoha/patches-6.12/029-13-spi-airoha-avoid-reading-flash-page-settings-from.patch
+++ b/target/linux/airoha/patches-6.12/029-13-spi-airoha-avoid-reading-flash-page-settings-from.patch
@@ -156,7 +156,7 @@ Signed-off-by: Mikhail Kshevetskiy <mikhail.kshevetskiy@iopsys.eu>
  	.supports_op = airoha_snand_supports_op,
  	.exec_op = airoha_snand_exec_op,
  	.dirmap_create = airoha_snand_dirmap_create,
-@@ -1105,36 +1030,6 @@ static int airoha_snand_setup(struct spi
+@@ -1110,36 +1035,6 @@ static int airoha_snand_setup(struct spi
  	return 0;
  }
  
@@ -193,7 +193,7 @@ Signed-off-by: Mikhail Kshevetskiy <mikhail.kshevetskiy@iopsys.eu>
  static const struct regmap_config spi_ctrl_regmap_config = {
  	.name		= "ctrl",
  	.reg_bits	= 32,
-@@ -1208,7 +1103,7 @@ static int airoha_snand_probe(struct pla
+@@ -1231,7 +1126,7 @@ static int airoha_snand_probe(struct pla
  	ctrl->setup = airoha_snand_setup;
  	device_set_node(&ctrl->dev, dev_fwnode(dev));
  
diff --git a/target/linux/airoha/patches-6.12/029-14-spi-airoha-snfi-en7523-workaround-flash-damaging.patch b/target/linux/airoha/patches-6.12/029-14-spi-airoha-snfi-en7523-workaround-flash-damaging.patch
deleted file mode 100644
index c40645b0021e41..00000000000000
--- a/target/linux/airoha/patches-6.12/029-14-spi-airoha-snfi-en7523-workaround-flash-damaging.patch
+++ /dev/null
@@ -1,97 +0,0 @@
-From 86f88e604305186aec1fb6eebbf8f0a42c2435d3 Mon Sep 17 00:00:00 2001
-From: Mikhail Kshevetskiy <mikhail.kshevetskiy@iopsys.eu>
-Date: Thu, 9 Oct 2025 19:33:23 +0300
-Subject: [PATCH] spi: airoha-snfi: en7523: workaround flash damaging
- if UART_TXD was short to GND
-
-We found that some serial console may pull TX line to GROUND during board
-boot time. Airoha uses TX line as one of it's BOOT pins. This will lead
-to booting in RESERVED boot mode.
-
-It was found that some flashes operates incorrectly in RESERVED mode.
-Micron and Skyhigh flashes are definitely affected by the issue,
-Winbond flashes are NOT affected.
-
-Details:
---------
-DMA reading of odd pages on affected flashes operates incorrectly. Page
-reading offset (start of the page) on hardware level is replaced by 0x10.
-Thus results in incorrect data reading. Usage of UBI make things even
-worse. Any attempt to access UBI leads to ubi damaging. As result OS loading
-becomes impossible.
-
-Non-DMA reading is OK.
-
-This patch detects booting in reserved mode, turn off DMA and print big
-fat warning.
-
-Signed-off-by: Mikhail Kshevetskiy <mikhail.kshevetskiy@iopsys.eu>
----
- drivers/spi/spi-airoha-snfi.c | 40 ++++++++++++++++++++++++++++++-----
- 1 file changed, 35 insertions(+), 5 deletions(-)
-
---- a/drivers/spi/spi-airoha-snfi.c
-+++ b/drivers/spi/spi-airoha-snfi.c
-@@ -1013,6 +1013,11 @@ static const struct spi_controller_mem_o
- 	.dirmap_write = airoha_snand_dirmap_write,
- };
- 
-+static const struct spi_controller_mem_ops airoha_snand_nodma_mem_ops = {
-+	.supports_op = airoha_snand_supports_op,
-+	.exec_op = airoha_snand_exec_op,
-+};
-+
- static int airoha_snand_setup(struct spi_device *spi)
- {
- 	struct airoha_snand_ctrl *as_ctrl;
-@@ -1058,7 +1063,8 @@ static int airoha_snand_probe(struct pla
- 	struct device *dev = &pdev->dev;
- 	struct spi_controller *ctrl;
- 	void __iomem *base;
--	int err;
-+	int err, dma_enabled;
-+	u32 sfc_strap;
- 
- 	ctrl = devm_spi_alloc_host(dev, sizeof(*as_ctrl));
- 	if (!ctrl)
-@@ -1092,12 +1098,36 @@ static int airoha_snand_probe(struct pla
- 		return dev_err_probe(dev, PTR_ERR(as_ctrl->spi_clk),
- 				     "unable to get spi clk\n");
- 
--	err = dma_set_mask(as_ctrl->dev, DMA_BIT_MASK(32));
--	if (err)
--		return err;
-+	dma_enabled = 1;
-+	if (device_is_compatible(dev, "airoha,en7523-snand")) {
-+		err = regmap_read(as_ctrl->regmap_ctrl,
-+				  REG_SPI_CTRL_SFC_STRAP, &sfc_strap);
-+		if (err)
-+			return err;
-+
-+		if (!(sfc_strap & 0x04)) {
-+			dma_enabled = 0;
-+			printk(KERN_WARNING "\n"
-+				"=== WARNING ======================================================\n"
-+				"Detected booting in RESERVED mode (UART_TXD was short to GND).\n"
-+				"This mode is known for incorrect DMA reading of some flashes.\n"
-+				"Usage of DMA for flash operations will be disabled to prevent data\n"
-+				"damage. Unplug your serial console and power cycle the board\n"
-+				"to boot with full performance.\n"
-+				"==================================================================\n\n");
-+		}
-+	}
-+
-+	if (dma_enabled) {
-+		err = dma_set_mask(as_ctrl->dev, DMA_BIT_MASK(32));
-+		if (err)
-+			return err;
-+	}
- 
- 	ctrl->num_chipselect = 2;
--	ctrl->mem_ops = &airoha_snand_mem_ops;
-+	ctrl->mem_ops = dma_enabled ?
-+				&airoha_snand_mem_ops :
-+				&airoha_snand_nodma_mem_ops;
- 	ctrl->bits_per_word_mask = SPI_BPW_MASK(8);
- 	ctrl->mode_bits = SPI_RX_DUAL;
- 	ctrl->setup = airoha_snand_setup;
diff --git a/target/linux/bcm27xx/patches-6.12/950-0073-ASoC-Add-support-for-all-the-downstream-rpi-sound-ca.patch b/target/linux/bcm27xx/patches-6.12/950-0073-ASoC-Add-support-for-all-the-downstream-rpi-sound-ca.patch
index 61b880bb43d6ab..3bf4122503f69b 100644
--- a/target/linux/bcm27xx/patches-6.12/950-0073-ASoC-Add-support-for-all-the-downstream-rpi-sound-ca.patch
+++ b/target/linux/bcm27xx/patches-6.12/950-0073-ASoC-Add-support-for-all-the-downstream-rpi-sound-ca.patch
@@ -15556,7 +15556,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  	imply SND_SOC_MAX98088
  	imply SND_SOC_MAX98090
  	imply SND_SOC_MAX98095
-@@ -175,6 +177,7 @@ config SND_SOC_ALL_CODECS
+@@ -176,6 +178,7 @@ config SND_SOC_ALL_CODECS
  	imply SND_SOC_PCM179X_SPI
  	imply SND_SOC_PCM186X_I2C
  	imply SND_SOC_PCM186X_SPI
@@ -15564,7 +15564,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  	imply SND_SOC_PCM3008
  	imply SND_SOC_PCM3060_I2C
  	imply SND_SOC_PCM3060_SPI
-@@ -267,6 +270,7 @@ config SND_SOC_ALL_CODECS
+@@ -268,6 +271,7 @@ config SND_SOC_ALL_CODECS
  	imply SND_SOC_TLV320ADCX140
  	imply SND_SOC_TLV320AIC23_I2C
  	imply SND_SOC_TLV320AIC23_SPI
@@ -15572,7 +15572,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  	imply SND_SOC_TLV320AIC26
  	imply SND_SOC_TLV320AIC31XX
  	imply SND_SOC_TLV320AIC32X4_I2C
-@@ -424,12 +428,12 @@ config SND_SOC_AD193X
+@@ -425,12 +429,12 @@ config SND_SOC_AD193X
  	tristate
  
  config SND_SOC_AD193X_SPI
@@ -15587,7 +15587,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  	depends on I2C
  	select SND_SOC_AD193X
  
-@@ -1229,6 +1233,13 @@ config SND_SOC_LOCHNAGAR_SC
+@@ -1230,6 +1234,13 @@ config SND_SOC_LOCHNAGAR_SC
  	  This driver support the sound card functionality of the Cirrus
  	  Logic Lochnagar audio development board.
  
@@ -15601,7 +15601,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  config SND_SOC_MADERA
  	tristate
  	default y if SND_SOC_CS47L15=y
-@@ -1638,6 +1649,10 @@ config SND_SOC_RT5616
+@@ -1639,6 +1650,10 @@ config SND_SOC_RT5616
  	tristate "Realtek RT5616 CODEC"
  	depends on I2C
  
@@ -15612,7 +15612,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  config SND_SOC_RT5631
  	tristate "Realtek ALC5631/RT5631 CODEC"
  	depends on I2C
-@@ -1995,6 +2010,9 @@ config SND_SOC_TFA9879
+@@ -1996,6 +2011,9 @@ config SND_SOC_TFA9879
  	tristate "NXP Semiconductors TFA9879 amplifier"
  	depends on I2C
  
@@ -15622,7 +15622,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  config SND_SOC_TFA989X
  	tristate "NXP/Goodix TFA989X (TFA1) amplifiers"
  	depends on I2C
-@@ -2596,4 +2614,8 @@ config SND_SOC_LPASS_TX_MACRO
+@@ -2601,4 +2619,8 @@ config SND_SOC_LPASS_TX_MACRO
  	select SND_SOC_LPASS_MACRO_COMMON
  	tristate "Qualcomm TX Macro in LPASS(Low Power Audio SubSystem)"
  
@@ -15633,7 +15633,7 @@ Signed-off-by: j-schambacher <joerg@hifiberry.com>
  endmenu
 --- a/sound/soc/codecs/Makefile
 +++ b/sound/soc/codecs/Makefile
-@@ -820,3 +820,12 @@ obj-$(CONFIG_SND_SOC_LPASS_TX_MACRO)	+=
+@@ -822,3 +822,12 @@ obj-$(CONFIG_SND_SOC_LPASS_TX_MACRO)	+=
  
  # Mux
  obj-$(CONFIG_SND_SOC_SIMPLE_MUX)	+= snd-soc-simple-mux.o
diff --git a/target/linux/bcm27xx/patches-6.12/950-0348-usb-dwc3-Set-DMA-and-coherent-masks-early.patch b/target/linux/bcm27xx/patches-6.12/950-0348-usb-dwc3-Set-DMA-and-coherent-masks-early.patch
index 97030141eb3576..d93a73f7f61216 100644
--- a/target/linux/bcm27xx/patches-6.12/950-0348-usb-dwc3-Set-DMA-and-coherent-masks-early.patch
+++ b/target/linux/bcm27xx/patches-6.12/950-0348-usb-dwc3-Set-DMA-and-coherent-masks-early.patch
@@ -351,7 +351,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  
 --- a/drivers/usb/dwc3/host.c
 +++ b/drivers/usb/dwc3/host.c
-@@ -126,10 +126,12 @@ out:
+@@ -129,10 +129,12 @@ out:
  
  int dwc3_host_init(struct dwc3 *dwc)
  {
@@ -364,7 +364,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  
  	/*
  	 * Some platforms need to power off all Root hub ports immediately after DWC3 set to host
-@@ -141,7 +143,12 @@ int dwc3_host_init(struct dwc3 *dwc)
+@@ -144,7 +146,12 @@ int dwc3_host_init(struct dwc3 *dwc)
  	if (irq < 0)
  		return irq;
  
diff --git a/target/linux/bcm27xx/patches-6.12/950-0380-irqchip-irq-brcmstb-l2-Add-config-for-2711-controlle.patch b/target/linux/bcm27xx/patches-6.12/950-0380-irqchip-irq-brcmstb-l2-Add-config-for-2711-controlle.patch
index b203e7a34185d1..5909f5d66bc4e2 100644
--- a/target/linux/bcm27xx/patches-6.12/950-0380-irqchip-irq-brcmstb-l2-Add-config-for-2711-controlle.patch
+++ b/target/linux/bcm27xx/patches-6.12/950-0380-irqchip-irq-brcmstb-l2-Add-config-for-2711-controlle.patch
@@ -52,7 +52,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
  /* L2 intc private data structure */
  struct brcmstb_l2_intc_data {
  	struct irq_domain *domain;
-@@ -299,11 +309,18 @@ static int __init brcmstb_l2_lvl_intc_of
+@@ -295,11 +305,18 @@ static int brcmstb_l2_lvl_intc_of_init(s
  	return brcmstb_l2_intc_of_init(np, parent, &l2_lvl_intc_init);
  }
  
diff --git a/target/linux/generic/hack-6.12/902-debloat_proc.patch b/target/linux/generic/hack-6.12/902-debloat_proc.patch
index cd7241e6afa012..239561c0e4d232 100644
--- a/target/linux/generic/hack-6.12/902-debloat_proc.patch
+++ b/target/linux/generic/hack-6.12/902-debloat_proc.patch
@@ -409,7 +409,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  
 --- a/net/ipv4/inet_timewait_sock.c
 +++ b/net/ipv4/inet_timewait_sock.c
-@@ -296,7 +296,7 @@ void __inet_twsk_schedule(struct inet_ti
+@@ -285,7 +285,7 @@ void __inet_twsk_schedule(struct inet_ti
  	 */
  
  	if (!rearm) {
diff --git a/target/linux/generic/kernel-6.12 b/target/linux/generic/kernel-6.12
index 0a9808b6bec272..463c22204a104f 100644
--- a/target/linux/generic/kernel-6.12
+++ b/target/linux/generic/kernel-6.12
@@ -1,2 +1,2 @@
-LINUX_VERSION-6.12 = .62
-LINUX_KERNEL_HASH-6.12.62 = 13e2c685ac8fab5dd992dd105732554dae514aef350c2a8c7418e7b74eb62c13
+LINUX_VERSION-6.12 = .63
+LINUX_KERNEL_HASH-6.12.63 = 9502c5ffe4b894383c97abfccf74430a84732f04ee476b9c0d87635b29df7db3
diff --git a/target/linux/generic/pending-6.12/700-netfilter-nft_flow_offload-handle-netdevice-events-f.patch b/target/linux/generic/pending-6.12/700-netfilter-nft_flow_offload-handle-netdevice-events-f.patch
index 301defb49a7719..59868b2a5352a3 100644
--- a/target/linux/generic/pending-6.12/700-netfilter-nft_flow_offload-handle-netdevice-events-f.patch
+++ b/target/linux/generic/pending-6.12/700-netfilter-nft_flow_offload-handle-netdevice-events-f.patch
@@ -55,7 +55,7 @@ Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
  }
 --- a/net/netfilter/nft_flow_offload.c
 +++ b/net/netfilter/nft_flow_offload.c
-@@ -487,47 +487,14 @@ static struct nft_expr_type nft_flow_off
+@@ -494,47 +494,14 @@ static struct nft_expr_type nft_flow_off
  	.owner		= THIS_MODULE,
  };
  
diff --git a/target/linux/qualcommax/patches-6.12/0113-remoteproc-qcom-Add-secure-PIL-support.patch b/target/linux/qualcommax/patches-6.12/0113-remoteproc-qcom-Add-secure-PIL-support.patch
index fc35733fa25844..6b11f048087509 100644
--- a/target/linux/qualcommax/patches-6.12/0113-remoteproc-qcom-Add-secure-PIL-support.patch
+++ b/target/linux/qualcommax/patches-6.12/0113-remoteproc-qcom-Add-secure-PIL-support.patch
@@ -25,7 +25,7 @@ Signed-off-by: Nikhil Prakash V <nprakash@codeaurora.org>
 @@ -86,6 +87,9 @@
  #define TCSR_WCSS_CLK_ENABLE	0x14
  
- #define MAX_HALT_REG		3
+ #define MAX_HALT_REG		4
 +
 +#define WCNSS_PAS_ID		6
 +
diff --git a/target/linux/qualcommax/patches-6.12/0116-remoteproc-qcom-Update-regmap-offsets-for-halt-regis.patch b/target/linux/qualcommax/patches-6.12/0116-remoteproc-qcom-Update-regmap-offsets-for-halt-regis.patch
deleted file mode 100644
index 7989dfa5c95669..00000000000000
--- a/target/linux/qualcommax/patches-6.12/0116-remoteproc-qcom-Update-regmap-offsets-for-halt-regis.patch
+++ /dev/null
@@ -1,79 +0,0 @@
-From 8c73af6e8d78c66cfef0f551b00d375ec0b67ff3 Mon Sep 17 00:00:00 2001
-From: Gokul Sriram Palanisamy <gokulsri@codeaurora.org>
-Date: Sat, 30 Jan 2021 10:50:09 +0530
-Subject: [PATCH] remoteproc: qcom: Update regmap offsets for halt register
-
-Fixed issue in reading halt-regs parameter from device-tree.
-
-Signed-off-by: Gokul Sriram Palanisamy <gokulsri@codeaurora.org>
-Signed-off-by: Sricharan R <sricharan@codeaurora.org>
----
- drivers/remoteproc/qcom_q6v5_wcss.c | 22 ++++++++++++++--------
- 1 file changed, 14 insertions(+), 8 deletions(-)
-
---- a/drivers/remoteproc/qcom_q6v5_wcss.c
-+++ b/drivers/remoteproc/qcom_q6v5_wcss.c
-@@ -86,7 +86,7 @@
- #define TCSR_WCSS_CLK_MASK	0x1F
- #define TCSR_WCSS_CLK_ENABLE	0x14
- 
--#define MAX_HALT_REG		3
-+#define MAX_HALT_REG		4
- 
- #define WCNSS_PAS_ID		6
- 
-@@ -155,6 +155,7 @@ struct wcss_data {
- 	u32 version;
- 	bool aon_reset_required;
- 	bool wcss_q6_reset_required;
-+	bool bcr_reset_required;
- 	const char *ssr_name;
- 	const char *sysmon_name;
- 	int ssctl_id;
-@@ -876,10 +877,13 @@ static int q6v5_wcss_init_reset(struct q
- 		}
- 	}
- 
--	wcss->wcss_q6_bcr_reset = devm_reset_control_get_exclusive(dev, "wcss_q6_bcr_reset");
--	if (IS_ERR(wcss->wcss_q6_bcr_reset)) {
--		dev_err(wcss->dev, "unable to acquire wcss_q6_bcr_reset\n");
--		return PTR_ERR(wcss->wcss_q6_bcr_reset);
-+	if (desc->bcr_reset_required) {
-+		wcss->wcss_q6_bcr_reset = devm_reset_control_get_exclusive(dev,
-+									   "wcss_q6_bcr_reset");
-+		if (IS_ERR(wcss->wcss_q6_bcr_reset)) {
-+			dev_err(wcss->dev, "unable to acquire wcss_q6_bcr_reset\n");
-+			return PTR_ERR(wcss->wcss_q6_bcr_reset);
-+		}
- 	}
- 
- 	return 0;
-@@ -929,9 +933,9 @@ static int q6v5_wcss_init_mmio(struct q6
- 		return -EINVAL;
- 	}
- 
--	wcss->halt_q6 = halt_reg[0];
--	wcss->halt_wcss = halt_reg[1];
--	wcss->halt_nc = halt_reg[2];
-+	wcss->halt_q6 = halt_reg[1];
-+	wcss->halt_wcss = halt_reg[2];
-+	wcss->halt_nc = halt_reg[3];
- 
- 	return 0;
- }
-@@ -1173,6 +1177,7 @@ static const struct wcss_data wcss_ipq80
- 	.crash_reason_smem = WCSS_CRASH_REASON,
- 	.aon_reset_required = true,
- 	.wcss_q6_reset_required = true,
-+	.bcr_reset_required = false,
- 	.ssr_name = "q6wcss",
- 	.ops = &q6v5_wcss_ipq8074_ops,
- 	.requires_force_stop = true,
-@@ -1187,6 +1192,7 @@ static const struct wcss_data wcss_qcs40
- 	.version = WCSS_QCS404,
- 	.aon_reset_required = false,
- 	.wcss_q6_reset_required = false,
-+	.bcr_reset_required = true,
- 	.ssr_name = "mpss",
- 	.sysmon_name = "wcnss",
- 	.ssctl_id = 0x12,
diff --git a/target/linux/qualcommax/patches-6.12/0119-remoteproc-wcss-disable-auto-boot-for-IPQ8074.patch b/target/linux/qualcommax/patches-6.12/0119-remoteproc-wcss-disable-auto-boot-for-IPQ8074.patch
index ccb78dec035d0d..842e289ec07ac3 100644
--- a/target/linux/qualcommax/patches-6.12/0119-remoteproc-wcss-disable-auto-boot-for-IPQ8074.patch
+++ b/target/linux/qualcommax/patches-6.12/0119-remoteproc-wcss-disable-auto-boot-for-IPQ8074.patch
@@ -14,7 +14,7 @@ Signed-off-by: Robert Marko <robimarko@gmail.com>
 
 --- a/drivers/remoteproc/qcom_q6v5_wcss.c
 +++ b/drivers/remoteproc/qcom_q6v5_wcss.c
-@@ -162,6 +162,7 @@ struct wcss_data {
+@@ -161,6 +161,7 @@ struct wcss_data {
  	const struct rproc_ops *ops;
  	bool requires_force_stop;
  	bool need_mem_protection;
@@ -22,7 +22,7 @@ Signed-off-by: Robert Marko <robimarko@gmail.com>
  };
  
  static int q6v5_wcss_reset(struct q6v5_wcss *wcss)
-@@ -1151,6 +1152,7 @@ static int q6v5_wcss_probe(struct platfo
+@@ -1147,6 +1148,7 @@ static int q6v5_wcss_probe(struct platfo
  						      desc->sysmon_name,
  						      desc->ssctl_id);
  
@@ -30,7 +30,7 @@ Signed-off-by: Robert Marko <robimarko@gmail.com>
  	ret = rproc_add(rproc);
  	if (ret)
  		return ret;
-@@ -1182,6 +1184,7 @@ static const struct wcss_data wcss_ipq80
+@@ -1177,6 +1179,7 @@ static const struct wcss_data wcss_ipq80
  	.ops = &q6v5_wcss_ipq8074_ops,
  	.requires_force_stop = true,
  	.need_mem_protection = true,
@@ -38,7 +38,7 @@ Signed-off-by: Robert Marko <robimarko@gmail.com>
  };
  
  static const struct wcss_data wcss_qcs404_res_init = {
-@@ -1198,6 +1201,7 @@ static const struct wcss_data wcss_qcs40
+@@ -1192,6 +1195,7 @@ static const struct wcss_data wcss_qcs40
  	.ssctl_id = 0x12,
  	.ops = &q6v5_wcss_qcs404_ops,
  	.requires_force_stop = false,
diff --git a/target/linux/qualcommax/patches-6.12/0136-remoteproc-qcom-wcss-populate-driver-data-for-IPQ601.patch b/target/linux/qualcommax/patches-6.12/0136-remoteproc-qcom-wcss-populate-driver-data-for-IPQ601.patch
index b1175d48e10fcf..142eefea9cf0d6 100644
--- a/target/linux/qualcommax/patches-6.12/0136-remoteproc-qcom-wcss-populate-driver-data-for-IPQ601.patch
+++ b/target/linux/qualcommax/patches-6.12/0136-remoteproc-qcom-wcss-populate-driver-data-for-IPQ601.patch
@@ -12,7 +12,7 @@ Signed-off-by: Gokul Sriram Palanisamy <gokulsri@codeaurora.org>
 
 --- a/drivers/remoteproc/qcom_q6v5_wcss.c
 +++ b/drivers/remoteproc/qcom_q6v5_wcss.c
-@@ -970,7 +970,7 @@ static int q6v5_alloc_memory_region(stru
+@@ -966,7 +966,7 @@ static int q6v5_alloc_memory_region(stru
  	return 0;
  }
  
@@ -21,7 +21,7 @@ Signed-off-by: Gokul Sriram Palanisamy <gokulsri@codeaurora.org>
  {
  	int ret;
  
-@@ -1173,7 +1173,7 @@ static void q6v5_wcss_remove(struct plat
+@@ -1169,7 +1169,7 @@ static void q6v5_wcss_remove(struct plat
  }
  
  static const struct wcss_data wcss_ipq8074_res_init = {
@@ -30,7 +30,7 @@ Signed-off-by: Gokul Sriram Palanisamy <gokulsri@codeaurora.org>
  	.q6_firmware_name = "IPQ8074/q6_fw.mdt",
  	.m3_firmware_name = "IPQ8074/m3_fw.mdt",
  	.crash_reason_smem = WCSS_CRASH_REASON,
-@@ -1187,6 +1187,20 @@ static const struct wcss_data wcss_ipq80
+@@ -1182,6 +1182,20 @@ static const struct wcss_data wcss_ipq80
  	.need_auto_boot = false,
  };
  
@@ -51,7 +51,7 @@ Signed-off-by: Gokul Sriram Palanisamy <gokulsri@codeaurora.org>
  static const struct wcss_data wcss_qcs404_res_init = {
  	.init_clock = qcs404_init_clock,
  	.init_regulator = qcs404_init_regulator,
-@@ -1206,6 +1220,7 @@ static const struct wcss_data wcss_qcs40
+@@ -1200,6 +1214,7 @@ static const struct wcss_data wcss_qcs40
  
  static const struct of_device_id q6v5_wcss_of_match[] = {
  	{ .compatible = "qcom,ipq8074-wcss-pil", .data = &wcss_ipq8074_res_init },
diff --git a/target/linux/qualcommax/patches-6.12/0905-remoteproc-q6v5_wcss-change-ssr-name-for-ipq6018-wif.patch b/target/linux/qualcommax/patches-6.12/0905-remoteproc-q6v5_wcss-change-ssr-name-for-ipq6018-wif.patch
index af7f45cc76309e..458418ba278f36 100644
--- a/target/linux/qualcommax/patches-6.12/0905-remoteproc-q6v5_wcss-change-ssr-name-for-ipq6018-wif.patch
+++ b/target/linux/qualcommax/patches-6.12/0905-remoteproc-q6v5_wcss-change-ssr-name-for-ipq6018-wif.patch
@@ -18,7 +18,7 @@ Signed-off-by: Mantas Pucka <mantas@8devices.com>
 
 --- a/drivers/remoteproc/qcom_q6v5_wcss.c
 +++ b/drivers/remoteproc/qcom_q6v5_wcss.c
-@@ -1143,9 +1143,9 @@ static int q6v5_wcss_probe(struct platfo
+@@ -1139,9 +1139,9 @@ static int q6v5_wcss_probe(struct platfo
  	if (ret)
  		return ret;
  
@@ -30,7 +30,7 @@ Signed-off-by: Mantas Pucka <mantas@8devices.com>
  
  	if (desc->ssctl_id)
  		wcss->sysmon = qcom_add_sysmon_subdev(rproc,
-@@ -1195,7 +1195,7 @@ static const struct wcss_data wcss_ipq60
+@@ -1190,7 +1190,7 @@ static const struct wcss_data wcss_ipq60
  	.aon_reset_required = true,
  	.wcss_q6_reset_required = true,
  	.bcr_reset_required = false,
diff --git a/target/linux/qualcommax/patches-6.12/0908-remoteproc-qcom_q6v5_wcss-add-optional-qdss_at-clock.patch b/target/linux/qualcommax/patches-6.12/0908-remoteproc-qcom_q6v5_wcss-add-optional-qdss_at-clock.patch
index ae2e500125beb8..d77573d63ffc60 100644
--- a/target/linux/qualcommax/patches-6.12/0908-remoteproc-qcom_q6v5_wcss-add-optional-qdss_at-clock.patch
+++ b/target/linux/qualcommax/patches-6.12/0908-remoteproc-qcom_q6v5_wcss-add-optional-qdss_at-clock.patch
@@ -21,7 +21,7 @@ Signed-off-by: Mantas Pucka <mantas@8devices.com>
  	struct regulator *cx_supply;
  	struct qcom_sysmon *sysmon;
  
-@@ -260,6 +261,9 @@ static int q6v5_wcss_start(struct rproc
+@@ -259,6 +260,9 @@ static int q6v5_wcss_start(struct rproc
  		return ret;
  	}
  
@@ -31,7 +31,7 @@ Signed-off-by: Mantas Pucka <mantas@8devices.com>
  	qcom_q6v5_prepare(&wcss->q6v5);
  
  	if (wcss->need_mem_protection) {
-@@ -773,6 +777,8 @@ static int q6v5_wcss_stop(struct rproc *
+@@ -772,6 +776,8 @@ static int q6v5_wcss_stop(struct rproc *
  	}
  
  pas_done:
@@ -40,7 +40,7 @@ Signed-off-by: Mantas Pucka <mantas@8devices.com>
  	clk_disable_unprepare(wcss->prng_clk);
  	qcom_q6v5_unprepare(&wcss->q6v5);
  
-@@ -981,6 +987,12 @@ static int ipq_init_clock(struct q6v5_wc
+@@ -977,6 +983,12 @@ static int ipq_init_clock(struct q6v5_wc
  			dev_err(wcss->dev, "Failed to get prng clock\n");
  		return ret;
  	}
diff --git a/target/linux/ramips/patches-6.12/850-awake-rt305x-dwc2-controller.patch b/target/linux/ramips/patches-6.12/850-awake-rt305x-dwc2-controller.patch
index 560a2efeacbab3..f963c756ec8b23 100644
--- a/target/linux/ramips/patches-6.12/850-awake-rt305x-dwc2-controller.patch
+++ b/target/linux/ramips/patches-6.12/850-awake-rt305x-dwc2-controller.patch
@@ -1,6 +1,6 @@
 --- a/drivers/usb/dwc2/platform.c
 +++ b/drivers/usb/dwc2/platform.c
-@@ -481,6 +481,12 @@ static int dwc2_driver_probe(struct plat
+@@ -484,6 +484,12 @@ static int dwc2_driver_probe(struct plat
  	if (retval)
  		return retval;
  
diff --git a/target/linux/realtek/patches-6.12/318-add-rtl83xx-clk-support.patch b/target/linux/realtek/patches-6.12/318-add-rtl83xx-clk-support.patch
index 72ac8a58c18360..2a580e670c59c9 100644
--- a/target/linux/realtek/patches-6.12/318-add-rtl83xx-clk-support.patch
+++ b/target/linux/realtek/patches-6.12/318-add-rtl83xx-clk-support.patch
@@ -23,7 +23,7 @@ Submitted-by: Markus Stockhausen <markus.stockhausen@gmx.de>
  source "drivers/clk/rockchip/Kconfig"
 --- a/drivers/clk/Makefile
 +++ b/drivers/clk/Makefile
-@@ -123,6 +123,7 @@ obj-$(CONFIG_COMMON_CLK_PISTACHIO)	+= pi
+@@ -122,6 +122,7 @@ obj-$(CONFIG_COMMON_CLK_PISTACHIO)	+= pi
  obj-$(CONFIG_COMMON_CLK_PXA)		+= pxa/
  obj-$(CONFIG_COMMON_CLK_QCOM)		+= qcom/
  obj-y					+= ralink/
diff --git a/target/linux/rockchip/patches-6.12/001-01-v6.13-arm64-dts-rockchip-Split-up-RK3588-s-PCIe-pinctrls.patch b/target/linux/rockchip/patches-6.12/001-01-v6.13-arm64-dts-rockchip-Split-up-RK3588-s-PCIe-pinctrls.patch
index 9a58adf4cd960e..6b1284f74c06ce 100644
--- a/target/linux/rockchip/patches-6.12/001-01-v6.13-arm64-dts-rockchip-Split-up-RK3588-s-PCIe-pinctrls.patch
+++ b/target/linux/rockchip/patches-6.12/001-01-v6.13-arm64-dts-rockchip-Split-up-RK3588-s-PCIe-pinctrls.patch
@@ -422,7 +422,7 @@ Signed-off-by: Heiko Stuebner <heiko@sntech.de>
  		};
 --- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
 +++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
-@@ -310,7 +310,7 @@
+@@ -311,7 +311,7 @@
  };
  
  &pcie2x1l2 {
@@ -431,7 +431,7 @@ Signed-off-by: Heiko Stuebner <heiko@sntech.de>
  	pinctrl-names = "default";
  	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
  	vpcie3v3-supply = <&vcc3v3_wf>;
-@@ -328,6 +328,10 @@
+@@ -329,6 +329,10 @@
  		pow_en: pow-en {
  			rockchip,pins = <0 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
  		};
diff --git a/target/linux/rockchip/patches-6.12/002-02-v6.13-arm64-dts-rockchip-Enable-HDMI0-on-rock-5a.patch b/target/linux/rockchip/patches-6.12/002-02-v6.13-arm64-dts-rockchip-Enable-HDMI0-on-rock-5a.patch
index c305197183bc14..8eaca4b9a7c3c8 100644
--- a/target/linux/rockchip/patches-6.12/002-02-v6.13-arm64-dts-rockchip-Enable-HDMI0-on-rock-5a.patch
+++ b/target/linux/rockchip/patches-6.12/002-02-v6.13-arm64-dts-rockchip-Enable-HDMI0-on-rock-5a.patch
@@ -37,7 +37,7 @@ Signed-off-by: Heiko Stuebner <heiko@sntech.de>
  	leds {
  		compatible = "gpio-leds";
  		pinctrl-names = "default";
-@@ -301,6 +313,31 @@
+@@ -302,6 +314,31 @@
  	status = "okay";
  };
  
@@ -69,7 +69,7 @@ Signed-off-by: Heiko Stuebner <heiko@sntech.de>
  &mdio1 {
  	rgmii_phy1: ethernet-phy@1 {
  		/* RTL8211F */
-@@ -793,3 +830,18 @@
+@@ -794,3 +831,18 @@
  &usb_host2_xhci {
  	status = "okay";
  };
diff --git a/target/linux/rockchip/patches-6.12/002-05-v6.15-arm64-dts-rockchip-Fix-label-name-of-hdptxphy-for-RK.patch b/target/linux/rockchip/patches-6.12/002-05-v6.15-arm64-dts-rockchip-Fix-label-name-of-hdptxphy-for-RK.patch
index 0e301958476d8f..1d901b83b219fc 100644
--- a/target/linux/rockchip/patches-6.12/002-05-v6.15-arm64-dts-rockchip-Fix-label-name-of-hdptxphy-for-RK.patch
+++ b/target/linux/rockchip/patches-6.12/002-05-v6.15-arm64-dts-rockchip-Fix-label-name-of-hdptxphy-for-RK.patch
@@ -15,7 +15,7 @@ Signed-off-by: Heiko Stuebner <heiko@sntech.de>
 
 --- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
 +++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
-@@ -334,7 +334,7 @@
+@@ -335,7 +335,7 @@
  	};
  };
  
diff --git a/target/linux/rockchip/patches-6.12/002-06-v6.15-arm64-dts-rockchip-Add-GPU-power-domain-regulator-de.patch b/target/linux/rockchip/patches-6.12/002-06-v6.15-arm64-dts-rockchip-Add-GPU-power-domain-regulator-de.patch
index c11bc86c9d5d99..76f6aa504c4dd7 100644
--- a/target/linux/rockchip/patches-6.12/002-06-v6.15-arm64-dts-rockchip-Add-GPU-power-domain-regulator-de.patch
+++ b/target/linux/rockchip/patches-6.12/002-06-v6.15-arm64-dts-rockchip-Add-GPU-power-domain-regulator-de.patch
@@ -35,7 +35,7 @@ Signed-off-by: Heiko Stuebner <heiko@sntech.de>
 
 --- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
 +++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5a.dts
-@@ -359,6 +359,10 @@
+@@ -360,6 +360,10 @@
  	status = "okay";
  };
  
diff --git a/target/linux/rockchip/patches-6.12/116-arm64-dts-rockchip-Update-LED-properties-for-Radxa-Ro.patch b/target/linux/rockchip/patches-6.12/116-arm64-dts-rockchip-Update-LED-properties-for-Radxa-Ro.patch
index 009c72992ce477..3cb14c23da4564 100644
--- a/target/linux/rockchip/patches-6.12/116-arm64-dts-rockchip-Update-LED-properties-for-Radxa-Ro.patch
+++ b/target/linux/rockchip/patches-6.12/116-arm64-dts-rockchip-Update-LED-properties-for-Radxa-Ro.patch
@@ -44,7 +44,7 @@ Signed-off-by: Tianling Shen <cnsztl@gmail.com>
  		};
  	};
  
-@@ -365,8 +375,9 @@
+@@ -366,8 +376,9 @@
  
  &pinctrl {
  	leds {
diff --git a/target/linux/siflower/patches-6.12/005-clk-add-drivers-for-siflower-socs.patch b/target/linux/siflower/patches-6.12/005-clk-add-drivers-for-siflower-socs.patch
index 4fb218d4821bdc..2be0427b79bc4e 100644
--- a/target/linux/siflower/patches-6.12/005-clk-add-drivers-for-siflower-socs.patch
+++ b/target/linux/siflower/patches-6.12/005-clk-add-drivers-for-siflower-socs.patch
@@ -20,7 +20,7 @@ Signed-off-by: Chuanhong Guo <gch981213@gmail.com>
  source "drivers/clk/sprd/Kconfig"
 --- a/drivers/clk/Makefile
 +++ b/drivers/clk/Makefile
-@@ -127,6 +127,7 @@ obj-y					+= renesas/
+@@ -126,6 +126,7 @@ obj-y					+= renesas/
  obj-$(CONFIG_ARCH_ROCKCHIP)		+= rockchip/
  obj-$(CONFIG_COMMON_CLK_SAMSUNG)	+= samsung/
  obj-$(CONFIG_CLK_SIFIVE)		+= sifive/
